test/test_nestedtensor.py

```diff
@@ -0,0 +1,173 @@
+# Owner(s): ["module: nestedtensor"]
+
+import torch
+from torch.testing._internal.common_utils import TestCase, load_tests
+from torch import nested_tensor
+
+# Tests are ported from pytorch/nestedtensor.
+# This makes porting as_nested_tensor easier in the future.
+def _iter_constructors():
+    # yield as_nested_tensor
+    yield nested_tensor
+
+
+class TestNestedTensor(TestCase):
+    @torch.inference_mode()
+    def test_unbind(self):
+        def _test_fn(unbind_fn):
+            def _test(a, b, c, d, e):
+                nt = nested_tensor([a, b])
+                a1, b1 = nt.unbind()
+                self.assertTrue(a is not a1)
+                self.assertTrue(b is not b1)
+
+                nt = nested_tensor([a, b], dtype=a.dtype)
+                a1, b1 = unbind_fn(nt, 0)
+                self.assertEqual(a, a1)
+                self.assertEqual(b, b1)
+
+                a = torch.randn((2, 3)).add_(1)
+                nt = nested_tensor([a])
+                self.assertEqual(a, unbind_fn(nt, 0)[0])
+
+            _test(
+                torch.tensor([1, 2]),
+                torch.tensor([7, 8]),
+                torch.tensor([3, 4]),
+                torch.tensor([5, 6]),
+                torch.tensor([6, 7]),
+            )
+            _test(
+                torch.tensor([1]),
+                torch.tensor([7]),
+                torch.tensor([3]),
+                torch.tensor([5]),
+                torch.tensor([6]),
+            )
+            _test(
+                torch.tensor(1),
+                torch.tensor(7),
+                torch.tensor(3),
+                torch.tensor(5),
+                torch.tensor(6),
+            )
+            _test(
+                torch.tensor([]),
+                torch.tensor([]),
+                torch.tensor([]),
+                torch.tensor([]),
+                torch.tensor([]),
+            )
```

Comment:
If you use subtests or parametrize, that will probably make for more pleasant test debugging if one particular test fails (cuz you can tell which subtest you were in without having to pdb)